### PR TITLE
Redis Cluster Support and Redis-py 4.1.0rc1

### PR DESCRIPTION
**Pull request by Andrew-Chen-Wang**
## What do these changes do?

Includes Redis cluster support (for this first commit) up to commit redis/redis-py@9db1eec71b443b8e7e74ff503bae651dc6edf411 so far

## Checklist

- [ ] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] If you provide code modification, please add yourself to `CONTRIBUTORS.txt`
  * The format is `<Name> <Surname>`.
  * Please keep alphabetical order, the file is sorted by names.
- [ ] Add a new news fragment into the [`CHANGES/`](../tree/master/CHANGES) folder
  * name it `<issue_id>.<type>` (e.g. `588.bugfix`)
  * if you don't have an `issue_id` change it to the pr id after creating the PR
  * ensure type is one of the following:
    * `.feature`: Signifying a new feature.
    * `.bugfix`: Signifying a bug fix.
    * `.doc`: Signifying a documentation improvement.
    * `.removal`: Signifying a deprecation or removal of public API.
    * `.misc`: A ticket has been closed, but it is not of interest to users.
  * Make sure to use full sentences with correct case and punctuation, for example:
   `Fix issue with non-ascii contents in doctest text files.`

### CODE REVIEW

File: aioredis/commands/parser.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    'COMMAND GETKEYS'.
+    """
+
+    def __init__(self, redis_connection):
+        self.initialized = False
+        self.commands = {}
+        self.initialize(redis_connection)
+
+    def initialize(self, r):
+        self.commands = r.execute_command("COMMAND")
```
Comment (Andrew-Chen-Wang) [inline aioredis/commands/parser.py:20]:
```suggestion
    async def initialize(self, r):
        self.commands = await r.execute_command("COMMAND")
```

**Comment by synodriver** [comment]
Wow, that's great.  ```redis-py``` can also benefit from it.

**Comment by apekkar** [comment]
Any news regarding this? Is it going to be included? It's been pending over a year already?

We see this is a big lacking and hope that this would get included...

**Comment by Andrew-Chen-Wang** [comment]
@apekkar this is now included in redis-py itself. It's just not well-documented. Please use redis-py's async portion. Thanks!

**Comment by apekkar** [comment]
Ok thanks for information!

And sorry for stupid question: but you are saying we should use redis-py for clustered mode instead of this library? You are not planning to add a support for RedisCluster?

**Pull request closed without merging**